src/Facade/Services/PurchaseOrderFacadeService.cs

```diff
@@ -344,50 +327,33 @@ public string GetOrderAsHtml(int orderNumber)
             return this.domainService.GetPurchaseOrderAsHtml(orderNumber);
         }
 
-        public new IResult<PurchaseOrderResource> Add(
-            PurchaseOrderResource resource, IEnumerable<string> privileges = null, int? userNumber = null)
+        protected override PurchaseOrder CreateFromResource(
+            PurchaseOrderResource resource,
+            IEnumerable<string> privileges = null)
         {
             var candidate = this.BuildEntityFromResourceHelper(resource);
 
-            PurchaseOrder order;
-            try
-            {
-                order = this.domainService.CreateOrder(candidate, privileges);
-            }
-            catch (DomainException exception)
-            {
-                this.logger.Info($"Failed to create order. {exception.Message}");
-                return new BadRequestResult<PurchaseOrderResource>(exception.Message);
-            }
+            var order = this.domainService.CreateOrder(candidate, privileges, out var makeCreditNote);
 
             this.transactionManager.Commit();
 
             this.domainService.CreateMiniOrder(order);
             this.transactionManager.Commit();
 
-            if (order.DocumentTypeName is "CO" or "RO")
+            if (!makeCreditNote)
             {
-                this.creditDebitNoteService.CreateDebitOrCreditNoteFromPurchaseOrder(order);
-                this.transactionManager.Commit();
+                return order;
             }
-            
-            order.Supplier = this.supplierRepository.FindById(order.SupplierId);
 
-            return new CreatedResult<PurchaseOrderResource>(
-                (PurchaseOrderResource)this.resourceBuilder.Build(order, privileges.ToList()));
-        }
+            this.creditDebitNoteService.CreateDebitOrCreditNoteFromPurchaseOrder(order);
+            this.transactionManager.Commit();
 
-        protected override PurchaseOrder CreateFromResource(
-            PurchaseOrderResource resource,
-            IEnumerable<string> privileges = null)
-        {
-            throw new NotImplementedException();
```

Comment:
And similiarly why the usual CreateFromResource as ignored / left unimplemented in favour of a 'new' .Add()?

The only thing I can think of is if whoever wrote this code wanted to prevent a transactionManager.Commit() happening at the end of the request...

Reply:
yeah no idea. a lot of this code is a bit weird. good that someone is starting to take a look at it.

Reply:
Ross made a good point that there was mini-order complication with this stuff

Reply:
@RossStewart anything I need to be wary of before I merge this?